src/HtmlProcessor/CssVariableEvaluator.php

```diff
@@ -0,0 +1,197 @@
+     *
+     * @param array<int, string> $matches
+     */
+    private function getPropertyValueReplacement(array $matches): string
+    {
+        $variableName = $matches[1];
+        if (isset($this->currentVariableDefinitions[$variableName])) {
+            return $this->currentVariableDefinitions[$variableName];
+        } else {
+            $fallbackValueSeparator = $matches[2] ?? '';
+            if ($fallbackValueSeparator !== '') {
+                $fallbackValue = $matches[3];
+                // The fallback value may use other CSS variables, so recurse
+                return $this->replaceVariablesInPropertyValue($fallbackValue);
+            } else {
+                return $matches[0];
+            }
+        }
+    }
+
+    /**
+     * Regular expression based on {@see https://stackoverflow.com/a/54143883/2511031 a StackOverflow answer}.
+     */
+    private function replaceVariablesInPropertyValue(string $propertyValue): string
+    {
+        return (new Preg())->replaceCallback(
+            '/
+                var\\(
+                    \\s*+
+                    # capture variable name including `--` prefix
+                    (
+                        --[^\\s\\),]++
+                    )
+                    \\s*+
+                    # capture optional fallback value
+                    (?:
+                        # capture separator to confirm there is a fallback value
+                        (,)\\s*
+                        # begin capture with named group that can be used recursively
+                        (?<recursable>
+                            # begin named group to match sequence without parentheses, except in strings
+                            (?<noparentheses>
+                                # repeated zero or more times:
+                                (?:
+                                    # sequence without parentheses or quotes
+                                    [^\\(\\)\'"]++
+                                    |
+                                    # string in double quotes
+                                    "(?>[^"\\\\]++|\\\\.)*"
+                                    |
+                                    # string in single quotes
+                                    \'(?>[^\'\\\\]++|\\\\.)*\'
+                                )*+
+                            )
+                            # repeated zero or more times:
+                            (?:
+                                # sequence in parentheses
+                                \\(
+                                    # using the named recursable pattern
+                                    (?&recursable)
+                                \\)
+                                # sequence without parentheses, except in strings
+                                (?&noparentheses)
+                            )*+
+                        )
+                    )?+
+                \\)
```

Comment:
Wow … see the master of regexp at work! :exploding_head: :+1:

Reply:
To be fair, this was mostly based on an [Stack Overflow answer](https://stackoverflow.com/questions/54143257/recursive-regex-for-matching-everything-in-parenthesis-pcre/54143883#54143883) by someone who is clearly much more expert, though I did fix a couple of issues with that.  The trick for handling quoted strings could certainly be useful elsewhere in the project where we are not yet using (or able to use) CssSelector or PHP-CSS-Parser - there are certainly some constructs that will currently fail.